sdk/search/Azure.Search/src/Generated/Operations/DocumentsClient.cs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core.Pipeline;
+using Azure.Search.Models;
+
+namespace Azure.Search
+{
+    internal partial class DocumentsClient
+    {
+        private readonly ClientDiagnostics clientDiagnostics;
+        private readonly HttpPipeline pipeline;
+        internal DocumentsRestClient RestClient { get; }
```

Comment:
This mirrors what I did in Storage, but I wonder if we should call this `DocumentsProtocol` and the property just `Protocol`.  Not urgent but something worth considering longer term if we want to consider exposing these as a super advanced escape hatch.

Reply:
Protocol sounds good too. I don't have a strong preference in any direction.